app/src/org/commcare/utils/UriToFilePath.java

```diff
@@ -135,4 +147,23 @@ private static boolean isDownloadsDocument(Uri uri) {
     private static boolean isMediaDocument(Uri uri) {
         return "com.android.providers.media.documents".equals(uri.getAuthority());
     }
+
+    /**
+     * @param uri The Uri to check.
+     * @return Whether the Uri authority is Google Photos.
+     */
+    private static boolean isGooglePhotosUri(Uri uri) {
+        return "com.google.android.apps.photos.content".equals(uri.getAuthority());
+    }
+
+    public static void grantPermissionForUri(Context context,Intent i, Uri uri, int flags) {
```

Comment:
Can you add a comment explaining what this does? Also I don't think this file is the best place for this method to live